ledger/src/advance.rs

```diff
@@ -71,7 +74,8 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
                 candidate_solutions,
                 candidate_transactions,
                 rng,
-            )?;
+            )
+            .with_context(|| "Failed to construct block template")?;
```

Comment:
```suggestion
            .with_context(|| "Failed to construct block")?;
```

Reply:
This is message is generated when `construct_block_template` fails. I think it is correct?